flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -1410,6 +1410,26 @@ object AggregateUtil {
         case _: SqlCountAggFunction =>
           aggregates(index) = new CountAggFunction
 
+        case collect: SqlAggFunction if collect.getKind == SqlKind.COLLECT =>
+          aggregates(index) = sqlTypeName match {
```

Comment:
We can pass the actual `TypeInformation` of the argument type here to the constructor of the `CollectAggFunction` and don't need to distinguish the different argument types.